front/src/modules/trainschedule/components/DriverTrainSchedule/DriverTrainScheduleExportCSV.ts

```diff
@@ -0,0 +1,148 @@
+import type { PositionSpeedTime, SpeedPosition, Train } from 'reducers/osrdsimulation/types';
+import * as d3 from 'd3';
+import { BaseOrEcoType } from './DriverTrainScheduleTypes';
+
+/**
+ * CSV Export of trainschedule
+ *
+ * Rows : position in km, speed in km/h, speed limit in km/h, time in s
+ *
+ */
+
+enum CSVKeys {
+  op = 'op',
+  ch = 'ch',
+  lineCode = 'lineCode',
+  trackName = 'trackName',
+  position = 'position',
+  speed = 'speed',
+  speedLimit = 'speedLimit',
+  time = 'time',
+}
+
+type CSVData = {
+  [key in keyof typeof CSVKeys]: string;
+};
+
+type PositionSpeedTimeOP = PositionSpeedTime & {
+  speedLimit?: string | number;
+  op?: '';
+  ch?: '';
+  lineCode?: '';
+  trackName?: '';
+};
+
+const pointToComma = (number: number) => number.toString().replace('.', ',');
+
+const interpolateValue = (
+  position: number,
+  speedList: PositionSpeedTime[],
+  value: 'speed' | 'time'
+): number => {
+  const bisector = d3.bisectLeft(
+    speedList.map((d: SpeedPosition) => d.position),
+    position
+  );
+  if (speedList[bisector - 1]) {
+    const totalDistance = speedList[bisector].position - speedList[bisector - 1].position;
+    const distance = position - speedList[bisector - 1].position;
+    const totalDifference = speedList[bisector][value] - speedList[bisector - 1][value];
+    return speedList[bisector - 1][value] + (totalDifference * distance) / totalDistance;
+  }
+  return speedList[bisector][value];
+};
+
+const getStepSpeedLimit = (position: number, speedLimitList: Train['vmax']) => {
+  const bisector = d3.bisectLeft(
+    speedLimitList.map((d: SpeedPosition) => d.position),
+    position
+  );
+  return speedLimitList[bisector].speed || 0;
```

Comment:
empty or zero will be the same on result graph, and it's more easy to do this this way because of type (we should spread an hybrid type accross all functions... I think we could close our eyes on this, it won't lead to an error, no?

Reply:
I have tested yesterday, if this function returns 0, here is what we have in the file at the end (0 in the speedLimit column). It's not very important, but it won't be the same that an empty cell. If it's ok for you, you can close this comment
![image](https://github.com/osrd-project/osrd/assets/45000526/2925263c-8737-43b9-99fa-79d20ab1bffd)